pandas-stubs/core/groupby/groupby.pyi

```diff
@@ -99,7 +99,7 @@ class GroupBy(BaseGroupBy[NDFrameT]):
     def last(self, **kwargs) -> DataFrame | Series: ...
     def max(self, **kwargs) -> DataFrame | Series: ...
     def min(self, **kwargs) -> DataFrame | Series: ...
-    def size(self) -> Series[int]: ...
+    def size(self) -> DataFrame | Series[int]: ...
```

Comment:
I'm not too familiar with the non-existing helper classes `pandas.core.groupby.generic._SeriesGroupBy(Non)Scalar/_DataFrameGroupBy(Non)Scalar`. It might be possible to avoid this union by having more complex overloads for `groupby(..., as_index=False)`. not sure whether that is worth it @Dr-Irv

Reply:
I'm not sure what to do here.  This is the same issue as #203 .  Ideally, we'd trace the use of `as_index=True` via some class manipulation to determine the right value of the result being `DataFrame` or `Series[int]` dependent on whether `as_index=True` was specified.  I started working on that about 6 months ago, but lost my working branch.  It's non-trivial.

The downside to the change proposed is that if people don't specify the `as_index` parameter (and I'm in that category of never using it because the default is `as_index=False`), then if you call `size()`, you will have to cast your result to `Series` when you know it is a `Series` because you used default arguments.

So let's say that the number of people using the default setting of `as_index=False` has A developers, and the number of people using `as_index=True` has B developers.  If we accept this PR, then all the A people have to make a code change and use `cast`.  If we don't accept this PR, then all the B people have to make a code change and use `cast` (the status quo).

I'm leaning towards not accepting it, because I think A is much larger than B, and leaving the issue open until we do it the "right" way, but my opinion can be changed.

Reply:
`as_index = True` is the default, no? And even though it's the default I would still contend that more people use it with `as_index = False`

Reply:
Ooops.  Got it backwards.  Have corrected it via editing.

Reply:
I don't really mind, I can always cast in my use-case where I'm using `as_index=False`

Reply:
> I don't really mind, I can always cast in my use-case where I'm using `as_index=False`

So should I close the PR?

Reply:
i've done it, thanks for your review and explanation!